
config:
  debug: false                    # for debug only
  odom_convert: true              # convert odom to posestamped using odom_convert_node
  monitor_node: true              # check recv rate
  warn_threshold: 3               # if recv rate smaller than max_freq-warn_threashold, then warn
  monitor_rate_hz: 500            # it must be greater than max_freq

services:
  # --------- Services ---------- #
  - srv_name: /add_two_ints
    srv_type: swarm_ros_bridge/AddTwoInts
    serverIp: drone0
    clientIp:
      - drone1
      - drone2
    srcPort: 2000
    prefix: true

topics:
# --------- Swarm Info ---------- #

- topic_name: /camera/color/image_raw
  msg_type: sensor_msgs/Image
  imgResizeRate: 1.0
  srcIP:
    - drone0
  srcPort: 3005
  max_freq: 10
  dstIP:
    - groundStation0
  prefix: false
  same_prefix: false

- topic_name: /camera/aligned_depth_to_color/image_raw
  msg_type: sensor_msgs/Image
  imgResizeRate: 1.0
  srcIP:
    - drone0
  srcPort: 3006
  max_freq: 10
  dstIP:
    - groundStation0
  prefix: false
  same_prefix: false

- topic_name: /ekf_quat/ekf_odom
  msg_type: nav_msgs/Odometry
  srcIP:
    - drone0
  srcPort: 3004
  max_freq: 50
  dstIP:
    - groundStation0
  prefix: false
  same_prefix: false


- topic_name: /goal_with_id_from_station
  msg_type: quadrotor_msgs/GoalSet
  srcIP:
    - groundStation0
  srcPort: 3003
  max_freq: -1
  dstIP:
    - drone0
  prefix: false
  same_prefix: false

- topic_name: /drone_0_ego_planner_node/grid_map/occupancy_inflate
  msg_type: sensor_msgs/PointCloud2
  cloudCompress: true
  cloudDownsample: -1.0
  srcIP:
    - drone0
  srcPort: 3007
  max_freq: 5
  dstIP:
    - groundStation0
  prefix: false
  same_prefix: false

- topic_name: /drone_0_ego_planner_node/optimal_list 
  msg_type: visualization_msgs/Marker
  srcIP:
    - drone0
  srcPort: 3008
  max_freq: 10
  dstIP:
    - groundStation0
  prefix: false
  same_prefix: false
  
- topic_name: /planning/ego_state_trigger
  msg_type: std_msgs/Bool
  srcIP:
    - drone0
  srcPort: 3009
  max_freq: -1
  dstIP:
    - groundStation0
  prefix: false
  same_prefix: false
